WordPress/Sniffs/PHP/FunctionEnqueuedSniff.php

```diff
@@ -0,0 +1,118 @@
+ * The Enqueued functions are:
+ * wp_register_script
+ * wp_enqueue_script
+ * wp_register_style
+ * wp_enqueue_style
+ * wp_register_style
+ *
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1146
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.14.0 This new sniff will check for a version in an Enqueued WP function
+ */
+class EnqueuedVersionSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 0.11.0
+	 * @var string
+	 */
+	protected $group_name = 'strict';
+
+	/**
+	 * List of array functions to which a $strict parameter can be passed.
+	 *
+	 * The $strict parameter is the third and last parameter for each of these functions.
+	 *
+	 * The array_keys() function only requires the $strict parameter when the optional
+	 * second parameter $search has been set.
+	 *
+	 * @link http://php.net/in-array
+	 * @link http://php.net/array-search
+	 * @link http://php.net/array-keys
+	 * @since 0.10.0
+	 * @since 0.11.0 Renamed from $array_functions to $target_functions.
+	 * @var array <string function_name> => <bool always needed ?>
+	 */
+	protected $target_functions = array(
+		'wp_register_script' => true,
+		'wp_enqueue_script'  => true,
+		'wp_register_style'  => true,
+		'wp_enqueue_style'   => true
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.11.0
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 * @return void
+	 */
+	public function process_parameters($stackPtr, $group_name, $matched_content, $parameters) {
+		// Check if the strict check is actually needed.
+		if ( false === $this->target_functions[$matched_content]) {
+			if (count($parameters) === 1) {
+				return;
+			}
+		}
+
+		// Check to see IF a source ($src) is specified otherwise its fine
+		if(isset($parameters[2])) {
+			/**
+			 * Version Check
+			 *
+			 * Check to make sure a Version ($ver) is set
+			 * Otherwise it will Show a warning to add a Src (url) to the Enqueued function
+			 */
+			if (false === isset($parameters[4]) || !$parameters[4]['raw']) {
+				$errorcode = 'MissingVersion';
+
+				$this->phpcsFile->addError(
+					'No Version found for %s; Please supply value for forth argument',
+					(isset( $parameters[4]['start']) ? $parameters[4]['start'] : $parameters[1]['start']),
+					$errorcode,
+					array($matched_content)
+				);
+				return;
+			}
+
+			/**
+			 * In footer Check
+			 *
+			 * Check to make sure that $in_footer is set to true
+			 * Otherwise it will warn the user to make sure if its correct
+			 */
+			if (false === isset($parameters[5]) || 'true' !== $parameters[5]['raw']) {
```

Comment:
Same comments as above for this whole snippet + I'm wondering whether if `false` is supplied as an explicit argument, the warning should still be thrown. In that case, it's a conscious choice of the dev to not enqueue in the footer.